fe/src/main/java/org/apache/doris/analysis/DateLiteral.java

```diff
@@ -169,36 +238,252 @@ protected void toThrift(TExprNode msg) {
         msg.date_literal = new TDateLiteral(getStringValue());
     }
 
-    public Date getValue() {
-        return date;
-    }
-
     @Override
     protected Expr uncheckedCastTo(Type targetType) throws AnalysisException {
         if (targetType.isDateType()) {
             return this;
         } else if (targetType.isStringType()) {
-            return new StringLiteral(getStringValue()); 
+            return new StringLiteral(getStringValue());
         }
         Preconditions.checkState(false);
         return this;
     }
 
+
+    private long makePackedDatetime() {
+        long ymd = ((year * 13 + month) << 5) | day;
+        long hms = (hour << 12) | (minute << 6) | second;
+        long packed_datetime = ((ymd << 17) | hms) << 24 + microsecond;
+        return packed_datetime;
+    }
+
     @Override
     public void write(DataOutput out) throws IOException {
         super.write(out);
-        out.writeLong(date.getTime());
+        out.writeLong(makePackedDatetime());
+    }
+
+    private void fromPackedDatetime(long packed_time) {
+        microsecond = (packed_time % (1L << 24));
+        long ymdhms = (packed_time >> 24);
+        long ymd = ymdhms >> 17;
+        long hms = ymdhms % (1 << 17);
+
+        day = ymd % (1 << 5);
+        long ym = ymd >> 5;
+        month = ym % 13;
+        year = ym / 13;
+        year %= 10000;
+        second = hms % (1 << 6);
+        minute = (hms >> 6) % (1 << 6);
+        hour = (hms >> 12);
+        this.type = Type.DATETIME;
     }
 
     @Override
     public void readFields(DataInput in) throws IOException {
         super.readFields(in);
-        date = new Date(in.readLong());
+        if (Catalog.getCurrentCatalogJournalVersion() >= FeMetaVersion.VERSION_60) {
+            fromPackedDatetime(in.readLong());
+        } else {
```

Comment:
I think we should persist type for later use. Or if we want to persist Date, we will have to change the format we use.